druid-shell/src/platform/x11/util.rs

```diff
@@ -69,17 +72,78 @@ pub fn refresh_rate(conn: &Rc<XCBConnection>, window_id: Window) -> Option<f64>
 }
 
 // Apparently you have to get the visualtype this way :|
-pub fn get_visual_from_screen(screen: &Screen) -> Option<Visualtype> {
+fn find_visual_from_screen(screen: &Screen, visual_id: u32) -> Option<Visualtype> {
     for depth in &screen.allowed_depths {
         for visual in &depth.visuals {
-            if visual.visual_id == screen.root_visual {
+            if visual.visual_id == visual_id {
                 return Some(*visual);
             }
         }
     }
     None
 }
 
+pub fn get_visual_from_screen(screen: &Screen) -> Option<Visualtype> {
+    find_visual_from_screen(screen, screen.root_visual)
+}
+
+pub fn get_argb_visual_type(
+    conn: &XCBConnection,
+    screen: &Screen,
+) -> Result<Option<Visualtype>, ReplyError> {
+    // Getting a visual is already funny, but finding the ARGB32 visual is even more fun.
+    // RENDER has picture formats. Each format corresponds to a visual. Thus, we first find the
+    // right picture format, then find the corresponding visual id, then the Visualtype.
+    if conn
+        .extension_information(render::X11_EXTENSION_NAME)?
+        .is_none()
+    {
+        // RENDER not supported
+        Ok(None)
+    } else {
+        let pict_formats = conn.render_query_pict_formats()?.reply()?;
+        // Find the ARGB32 standard format
+        let res = pict_formats
+            .formats
+            .iter()
+            .find(|format| {
+                format.type_ == render::PictType::DIRECT
+                    && format.depth == 32
+                    && format.direct.red_shift == 16
+                    && format.direct.red_mask == 0xff
+                    && format.direct.green_shift == 8
+                    && format.direct.green_mask == 0xff
+                    && format.direct.blue_shift == 0
+                    && format.direct.blue_mask == 0xff
+                    && format.direct.alpha_shift == 24
+                    && format.direct.alpha_mask == 0xff
+            })
+            // Now find the corresponding visual ID
+            .and_then(|format| {
+                pict_formats
+                    .screens
+                    .iter()
+                    .filter_map(|screen| {
```

Comment:
not sure why clippy is not warning about `filter_map` and `next`. You can use `find_map`